syft/workers/base.py

```diff
@@ -1263,3 +1263,16 @@ def force_detail(worker: AbstractWorker, worker_tuple: tuple) -> tuple:
                 worker.object_store.rm_obj(obj.id)
 
         return result
+
+    @classmethod
+    def is_framework_supported(cls, framework: str) -> bool:
+        """
+        Returns True if framework is supported, else returns False.
+        :param framework:
+        :return: True/False
+        """
+        from syft.frameworks import __all__
```

Comment:
Don’t we have a variable with the supported frameworks other than this `__all__`?
Also, is there a reason for this import to be inside the method?

Reply:
Currently, the list of supported frameworks in maintained in the __all__ variables, which is a list, in the framework package. Since  __all__  is a global list, importing it outside the func, will override the __all__ variable in that file.

Reply:
What about `from syft.generic.frameworks import framework_packages` at the top of the file?

Reply:
@Jasopaum but `framework_packages` doesn't include `keras` , don't we support keras??

Reply:
Maybe we can keras in the `framework_packages` in `syft/generic/framewroks/types.py` then?

Reply:
Do we support keras? Where does the code for that live?

Reply:
There are some files in `syft/frameworks/keras`

Reply:
Huh, I'm surprised that wasn't extracted to the repo for Tensorflow support, since it depends on `tensorflow-encrypted`.

Reply:
@Jasopaum have added keras in the framework_packages.

Reply:
@karlhigley tests are gettings timed out. They are running fine in the local. Can you help ?

Reply:
Re-running the tests